clone.go

```diff
@@ -0,0 +1,262 @@
+	}
+
+	// If, on a linear sequence, the last overhang's position plus EnzymeSkip is over the length of the sequence, remove that overhang.
+	if (seq.Circular == false) && (overhangs[len(overhangs)-1].Position+enzyme.EnzymeSkip > len(sequence)) {
+		overhangs = overhangs[:len(overhangs)-1]
+	}
+
+	// Sort overhangs
+	sort.SliceStable(overhangs, func(i, j int) bool {
+		return overhangs[i].Position < overhangs[j].Position
+	})
+
+	// Convert Overhangs into Fragments
+	var currentOverhang Overhang
+	var nextOverhang Overhang
+	if len(overhangs) == 1 { // Check the case of a single cut
+		if seq.Circular == true {
+			fragmentSeqs = append(fragmentSeqs, sequence[overhangs[0].Position:]+sequence[:overhangs[0].Position])
+		} else {
+			fragmentSeqs = append(fragmentSeqs, sequence[overhangs[0].Position:])
+			fragmentSeqs = append(fragmentSeqs, sequence[:overhangs[0].Position])
+		}
+	} else {
+		// The following will iterate over the overhangs list to turn them into fragments
+		// There are two important variables: if the sequence is circular, and if the enzyme cutting is directional. All Type IIS enzymes
+		// are directional, and in normal GoldenGate reactions these fragments would be constantly cut with enzyme as the reaction runs,
+		// so are removed from the output sequences. If the enzyme is not directional, all fragments are valid.
+		// If the sequence is circular, there is a chance that the nextOverhang's position will be greater than the length of the original sequence.
+		// This is ok, and represents a valid cut/fragmentation of a rotation of the sequence. However, everything after will be a repeat fragment
+		// of current fragments, so the iteration is terminated.
+		for overhangIndex := 0; overhangIndex < len(overhangs)-1; overhangIndex++ {
+			currentOverhang = overhangs[overhangIndex]
+			nextOverhang = overhangs[overhangIndex+1]
+			if enzyme.Directional == true {
+				if (currentOverhang.Forward == true) && (nextOverhang.Forward == false) {
+					fragmentSeqs = append(fragmentSeqs, sequence[currentOverhang.Position:nextOverhang.Position])
+				}
+				if nextOverhang.Position > len(seq.Sequence) {
+					break
+				}
+			} else {
+				fragmentSeqs = append(fragmentSeqs, sequence[currentOverhang.Position:nextOverhang.Position])
+				if nextOverhang.Position > len(seq.Sequence) {
+					break
+				}
+			}
+		}
+	}
+
+	// Convert fragment sequences into fragments
+	var fragments []Fragment
+	for _, fragment := range fragmentSeqs {
+		fragments = append(fragments, Fragment{Sequence: fragment[enzyme.EnzymeOverhangLen : len(fragment)-enzyme.EnzymeOverhangLen], ForwardOverhang: fragment[:enzyme.EnzymeOverhangLen], ReverseOverhang: fragment[len(fragment)-enzyme.EnzymeOverhangLen:]})
+	}
+	return fragments
+}
+
+func recurseLigate(wg *sync.WaitGroup, c chan string, seedFragment Fragment, fragmentList []Fragment) {
```

Comment:
```suggestion
func generateAllPossibleLigationsForFragments(wg *sync.WaitGroup, c chan string, seedFragment Fragment, fragmentList []Fragment) {
```

Reply:
Can we get a unit test for this function?

Also is there reference for this in case someone needs to fix this and you aren't around?

Reply:
Let me write some docs for it.